lib/std/crypto/tls/Client.zig

```diff
@@ -452,11 +452,20 @@ pub fn init(stream: anytype, ca_bundle: Certificate.Bundle, host: []const u8) In
                         if (ciphertext.len > cleartext_buf.len) return error.TlsRecordOverflow;
                         const cleartext = cleartext_buf[0..ciphertext.len];
                         const auth_tag = record_decoder.array(P.AEAD.tag_length).*;
-                        const V = @Vector(P.AEAD.nonce_length, u8);
-                        const pad = [1]u8{0} ** (P.AEAD.nonce_length - 8);
-                        const operand: V = pad ++ @as([8]u8, @bitCast(big(read_seq)));
+                        const nonce = if (builtin.zig_backend == .stage2_x86_64 and
+                            P.AEAD.nonce_length > comptime std.simd.suggestVectorLength(u8) orelse 1)
+                        nonce: {
+                            var nonce = p.server_handshake_iv;
+                            const operand = std.mem.readInt(u64, nonce[nonce.len - 8 ..], .big);
+                            std.mem.writeInt(u64, nonce[nonce.len - 8 ..], operand ^ read_seq, .big);
+                            break :nonce nonce;
+                        } else nonce: {
+                            const V = @Vector(P.AEAD.nonce_length, u8);
+                            const pad = [1]u8{0} ** (P.AEAD.nonce_length - 8);
+                            const operand: V = pad ++ @as([8]u8, @bitCast(big(read_seq)));
+                            break :nonce @as(V, p.server_handshake_iv) ^ operand;
+                        };
```

Comment:
The code before was so nice - certainly much easier to read and verify. We don't have any test coverage for this code yet either.

Can we please have vectors in the x86 backend instead? I can work on that if necessary. There is already too much `if (stage2_x86_64)` for my taste.

Reply:
The issue is that `@Vector(32, u8)` is not a native vector size without AVX, and non-native vectors are way down on the x86_64 backend TODO list.  To be clear, the `else` code path is the same code as it was before and `-fno-llvm -mcpu=x86_64_v3` currently compiles the `else` branch for every `nonce_length` in use just fine, where `-fno-llvm -mcpu=x86_64` only supports some `nonce_length`s in use.  I _really_ don't want to lose the "test coverage" of compiling this code with the self-hosted backend (in the sense that I do this all the time when using the self-hosted accelerated development cycle) just to avoid some extra code that I frequently keep an eye on by grepping for `stage2_x86_64`, and which can easily be reverted by just keeping the else branch.  Personally, I also find this fallback code easier to read, and would expect it to run faster (since one-off vector operations generally can't benefit from the increased throughput given that there is only one of them, and so you are only hit with the increased latency).

Reply:
Ok you convinced me